### ISSUE TITLE
Configurable option for file or migration

### ISSUE THREAD

**Issue by FoksVHox** [nWidart/laravel-modules#1084]
Hi! 
Let me start off by saying that this package is **VERY** helpful. However, I'm seeing a little problem with the `modules_statuses.json`. The issue that I'm facing is that is it hard to reach for customization on a product, like a shop system e.g. Blesta. I'd like to see a config option for having the module statuses in a database table or a file like `modules_statuses.json`

**Comment by nWidart** [comment]
Hi,

Thanks for the kind words 🙏 
Indeed the database driver would be nice, I started some work on it at some point here: https://github.com/nWidart/laravel-modules/tree/feature/database-modules
Never had time to finish it though. But the interfaces are there to implement.

**Comment by FoksVHox** [comment]
> Hi,
> 
> Thanks for the kind words 🙏
> Indeed the database driver would be nice, I started some work on it at some point here: https://github.com/nWidart/laravel-modules/tree/feature/database-modules
> Never had time to finish it though. But the interfaces are there to implement.

I'll poke around with it either tomorrow or on Wednesday, and might make a pr at the end of the week.

**Comment by FoksVHox** [comment]
So I looked into it. but it is so old that I think it would make more sense to create a new branch and start over. I'm not the most experienced person when it comes to developing and maintaining packages. @nWidart I'd prefer if you could finish the feature.

**Comment by FoksVHox** [comment]
> So I looked into it. but it is so old that I think it would make more sense to create a new branch and start over. I'm not the most experienced person when it comes to developing and maintaining packages.

**Comment by FoksVHox** [comment]
@nWidart Isn't this also some sort of feature request? (label)

**Comment by nWidart** [comment]
Why not, but I would prefer to close this issue as it will otherwise stay open forever :D 
This feature is rarely requested, and people who do request it don't want to implement it.
I personally don't have time for it currently.

**Comment by FoksVHox** [comment]
> Why not, but I would prefer to close this issue as it will otherwise stay open forever :D
> This feature is rarely requested, and people who do request it don't want to implement it.
> I personally don't have time for it currently.

@nWidart I understand that. I would do it myself, but I don't have the experience with making packages for Laravel with Composer. And I see why you would prefer to close this issue, and I know that a lot of people would just ask for a feature and expect that you will make it, but I'll try to make the feature next week.

**Comment by nWidart** [comment]
Yeah sorry, I wasn't directed at you specifically. I understand it's not as easy. And you're right my branch is quite old, but I'm pretty confident it should be ok to merge master back onto it and go from there.

**Comment by FoksVHox** [comment]
> Yeah sorry, I wasn't directed at you specifically. I understand it's not as easy. And you're right my branch is quite old, but I'm pretty confident it should be ok to merge master back onto it and go from there.

@nWidart I'll give it a go

**Comment by FoksVHox** [comment]
@nWidart any news?

**Comment by nWidart** [comment]
There is a pull request open I believe which adds the database driver.

**Comment by FoksVHox** [comment]
There is, it has been there for some time. Any clue when it'll be merged?
